### ISSUE TITLE
Investigate adding `auto-walrus` to linting checks

### ISSUE THREAD

**Issue by AetherUnbound** [WordPress/openverse-catalog#858]
## Description

We should investigate to see if [`auto-walrus`](https://github.com/MarcoGorelli/auto-walrus) can be added to the project's pre-commit & linting checks, and whether or not it'd be beneficial to have it as part of our checks.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

**Comment by dhruvkb** [comment]
This issue should go in `WordPress/openverse`. It can be beneficial in the API as well!

**Comment by AetherUnbound** [comment]
Great point! Although we're not syncing the pre-commit config across repos so having this in the main repo might make it harder to attach work to it, should I create a sibling issue in the API repo instead?

**Comment by sarayourfriend** [comment]
I would really strongly recommend against `auto-walrus`. The walrus operator can be great for readability sometimes, but in my experience it is often not a boon to readability and makes what are otherwise very simple conditional checks a massive pain to read and understand. I'm not against using the walrus operator but it is not a code-style choice that I think is so imperative to consistency that we would want to automatically apply it. In particular, the tool seems pretty niche and the ignore pragma is obscure and non-obvious (`# type: ignore`).

**Comment by krysal** [comment]
I agree with @sarayourfriend and would prefer not to automatically apply the `walrus` operator and let each programmer (and ultimately the team) decide when is best to use it.

**Comment by dhruvkb** [comment]
Since the discussion is going along the same lines as it did in the Python community when the operator was being introduced with people divided over readability and utility of adding a new operator, we can take the call along the same lines and let it be a personal choice to use or avoid it.

Personally I like the walrus operator and try to use it wherever I can, though I often forget about it, and would likely find a lint check handy.

**Comment by AetherUnbound** [comment]
@dhruvkb I appreciate your perspective on this! do you have any suggestions for making a lint check optional? Otherwise I think it might be best to closed this ticket and let folks decide on walrus operator usage on their own 🙂

**Comment by dhruvkb** [comment]
There is a [roundabout way](https://stackoverflow.com/a/59745587) to make a pre-commit check optional, but the author recommends not to use it.

**Comment by AetherUnbound** [comment]
Thanks! I think with that in mind I'll go ahead and close this then - thank you all for sharing your thoughts 🙌🏼
